torch/testing/_asserts.py

```diff
@@ -245,118 +264,350 @@ def _check_values_close(
+        check_device=check_device,
+        check_dtype=check_dtype,
+        check_stride=check_stride,
+    )
+    exc = _check_by_type(a, b, check_tensors)
     if exc:
         raise exc
 
 
-def assert_tensors_close(
-    a: torch.Tensor,
-    b: torch.Tensor,
+def assert_close(
+    a: Union[Tensor, Sequence[Tensor], Mapping[Any, Tensor]],
+    b: Union[Tensor, Sequence[Tensor], Mapping[Any, Tensor]],
     *,
     rtol: Optional[float] = None,
     atol: Optional[float] = None,
     check_device: bool = True,
     check_dtype: bool = True,
     check_stride: bool = True,
 ) -> None:
-    """Asserts that the values of two tensors are close up to a desired tolerance.
+    r"""Asserts that the values of tensors are close.
+
+    Closeness is defined by
+
+    .. math::
+
+        \lvert a - b \rvert \le \texttt{atol} + \texttt{rtol} \cdot \lvert b \rvert
 
     If both tolerances, :attr:`rtol` and :attr:`rtol`, are ``0``, asserts that :attr:`a` and :attr:`b` are bitwise
-    equal. Optionally, checks that some attributes of both tensors are equal.
+    equal.
+
+    Optionally, checks that some attributes of tensors are equal.
+
+    Also supports :class:`~collections.abc.Sequence`'s and :class:`~collections.abc.Mapping`'s of :class:`Tensor`'s.
 
     Args:
-        a (torch.Tensor): First tensor.
-        b (torch.Tensor): Second tensor.
+        a (Union[Tensor, Sequence[Tensor], Mapping[Any, Tensor]]): First input.
+        b (Union[Tensor, Sequence[Tensor], Mapping[Any, Tensor]]): Second input.
         rtol (Optional[float]): Relative tolerance. If specified :attr:`atol` must also be specified. If omitted,
             default values based on the :attr:`~torch.Tensor.dtype` are selected with the below table.
         atol (Optional[float]): Absolute tolerance. If specified :attr:`rtol` must also be specified. If omitted,
             default values based on the :attr:`~torch.Tensor.dtype` are selected with the below table.
-        check_device (bool): If ``True`` (default), asserts that both :attr:`a` and :attr:`b` live in the same
-            :attr:`~torch.Tensor.device` memory. If this check is disabled **and** :attr:`a` and :attr:`b` do not live
-            in the same memory :attr:`~torch.Tensor.device`, they are moved CPU memory before their values are
-            compared.
-        check_dtype (bool): If ``True`` (default), asserts that both :attr:`a` and :attr:`b` have the same
-            :attr:`~torch.Tensor.dtype`. If this check is disabled **and** :attr:`a` and :attr:`b` do not have the same
-            :attr:`~torch.Tensor.dtype`, they are copied to the :class:`~torch.dtype` returned by
-            :func:`torch.promote_types` before their values are compared.
-        check_stride (bool): If ``True`` (default), asserts that both :attr:`a` and :attr:`b` have the same stride.
+        check_device (bool): If ``True`` (default), asserts that tensors live in the same :attr:`~torch.Tensor.device`
```

Comment:
nit: "that both tensors are on the same device" would be more consistent with other PyTorch language

Reply:
(because tensors, like vampires, aren't really alive)